examples/text-generation/generation_gemma.py

```diff
@@ -56,7 +56,7 @@ def main():
     model_id = "google/gemma-2b"
     torch_dtype = torch.bfloat16
 
-    model = TpuModelForCausalLM.from_pretrained(model_id, torch_dtype=torch_dtype)
+    model = AutoModelForCausalLM.from_pretrained(model_id, torch_dtype=torch_dtype)
```

Comment:
Do we need the `torch_dtype=torch_dtype`? It should be taken from the config no?

Reply:
Well, it doesn't look like it works this way:
```python
>>> from transformers import AutoModelForCausalLM
model = AutoModelForCausalLM.from_pretrained("google/gemma-2b")
print(model.config.torch_dtype)
print(model.model.layers[0].self_attn.o_proj.weight.dtype)
>>> model = AutoModelForCausalLM.from_pretrained("google/gemma-2b")
Gemma's activation function should be approximate GeLU and not exact GeLU.
Changing the activation function to `gelu_pytorch_tanh`.if you want to use the legacy `gelu`, edit the `model.config` to set `hidden_activation=gelu`   instead of `hidden_act`. See https://github.com/huggingface/transformers/pull/29402 for more details.
Loading checkpoint shards: 100%|██████████████████████| 2/2 [00:00<00:00,  2.65it/s]
>>> print(model.config.torch_dtype)
torch.bfloat16
>>> print(model.model.layers[0].self_attn.o_proj.weight.dtype)
torch.float32
```